packages/plugin-npm-cli/sources/commands/npm/login.ts

```diff
@@ -0,0 +1,162 @@
+import {Configuration, MessageName}                     from '@berry/core';
+import {PluginConfiguration, StreamReport}              from '@berry/core';
+import {npmConfigUtils, npmHttpUtils}                   from '@berry/plugin-npm';
+import inquirer                                         from 'inquirer';
+import {Readable, Writable}                             from 'stream';
+import {URL}                                            from 'url';
+
+const SUCCESS_MESSAGE = `Successfully logged in`;
+const ERROR_MESSAGE = `Invalid Authentication`;
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: any, pluginConfiguration: PluginConfiguration) => clipanion
+
+  .command(`npm login [-r,--registry REGISTRY]`)
```

Comment:
What do you think of replacing `[-r,--registry]` with `[-s,--scope]`, to mirror the interface of `whoami`? This would also allow us not to have to complexify more the `npmHttpUtils` function to add support for random registries (maybe we eventually will, but I'd like to keep things simple for now).